### PR TITLE
fix(partition): deprecate partition operator and create partition sta…

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:**
Deprecate the current `partition` operator and create a new creation function partition observable.

**Related issue (if exists):** 
#4419

### CODE REVIEW

**Comment by dkosasih** [comment]
I am still working on the tests.
if the partition code can be reviewed meanwhile, that would be awesome :)

File: src/internal/observable/partition.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * returns true. The second Observable emits source values for which the
+ * predicate returns false. The first behaves like {@link filter} and the second
+ * behaves like {@link filter} with the predicate negated.
+ *
+ * ## Example
+ * Partition a set of numbers into odds and evens observables
+ * ```ts
+ * import { partition } from 'rxjs';
+ *
+ * const observableValues = of(1, 2, 3, 4, 5, 6);
```
Comment (dzhavat) [inline src/internal/observable/partition.ts]: Missing import for `of`.

File: src/internal/observable/partition.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * const [evens$, odds$] = partition(observableValues, (value, index) => value % 2 === 0);
+ *
+ * odds$.subscribe(x => console.log('odds', x));
+ * evens$.subscribe(x => console.log('evens', x));
+ * ```
```
Comment (dzhavat) [inline src/internal/observable/partition.ts]: It'd be nice to have the outputs of `odds$` and `evens$`.

File: src/internal/observable/partition.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * `false` the value is emitted on the second Observable in the array. The
+ * `index` parameter is the number `i` for the i-th source emission that has
+ * happened since the subscription, starting from the number `0`.
+ * @param {any} [thisArg] An optional argument to determine the value of `this`
+ * in the `predicate` function.
+ * @return {[Observable<T>, Observable<T>]} An array with two Observables: one
+ * with values that passed the predicate, and another with values that did not
+ * pass the predicate.
+ * @method partition
+ * @owner Observable
```
Comment (benlesh) [inline src/internal/observable/partition.ts]: `@method` and `@owner` can be removed. That's legacy from our old docs system.

**Comment by benlesh** [comment]
(I can see it's draft, but it's looking good so far)

**Comment by dkosasih** [comment]
It seems like there are few dtslint tests that are failing. 
Looks like it's related to inference of `{}` has changed to `unknown`.
should this be in a separate issue @benlesh ?

**Pull request merged**